CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 2.8)
+cmake_minimum_required (VERSION 2.8.12)
```

Comment:
@geoffromer, can you detect what version of CMAKE is being used and put the INTERFACE in there if it is available?

Something like:

```
target_link_libraries(benchmark ${CMAKE_TARGET_LINK_INTERFACE} ${CMAKE_THREAD_LIBS_INIT})
```

`CMAKE_TARGET_LINK_INTERFACE` would be empty if cmake is too old. Just thinking out loud - then it would do the right thing if the user has a newer version of cmake but gracefully fallback if not.